### PR TITLE
tests/robustness: Implement kubernetes list watch protocol

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor @jmhbnz @chaochn47

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM with minor comments.

File: tests/robustness/traffic/client.go [hunk 1]
```diff
@@ -179,32 +195,53 @@ func (c *RecordingClient) LeaseGrant(ctx context.Context, ttl int64) (int64, err
...
+
+func (c *RecordingClient) Watch(ctx context.Context, key string, rev int64, withPrefix bool) clientv3.WatchChan {
+	ops := []clientv3.OpOption{clientv3.WithProgressNotify()}
+	if withPrefix {
+		ops = append(ops, clientv3.WithPrefix())
+	}
+	if rev != 0 {
+		ops = append(ops, clientv3.WithRev(rev))
+	}
+	resp := make(chan clientv3.WatchResponse)
```
[the remark this answers is not in the document]
Comment (serathius) [reply tests/robustness/traffic/client.go:236]: Done

File: tests/robustness/traffic/kubernetes.go [hunk 2]
```diff
@@ -61,58 +65,97 @@ const (
...
+					continue
+				}
+				s.Reset(resp)
+				limiter.Wait(ctx)
+			}
+			err := t.Write(ctx, c, ids, s)
+			lastWriteFailed = err != nil
+			if err != nil {
+				continue
+			}
```
Comment (ahrtr) [inline tests/robustness/traffic/kubernetes.go:117]:
```suggestion
			if err := t.Write(ctx, c, ids, s); err != nil {
				lastWriteFailed = true
				continue
			}
```

File: tests/robustness/traffic/kubernetes.go [hunk 3]
```diff
@@ -61,58 +65,97 @@ const (
...
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case <-finish:
+				return nil
+			default:
+			}
+			// Avoid multiple failed writes in a row
+			if lastWriteFailed {
+				resp, err := t.Range(ctx, c, keyPrefix, true)
```
Comment (ahrtr) [inline tests/robustness/traffic/kubernetes.go:106]:
```suggestion
				lastWriteFailed = false
				resp, err := t.Range(ctx, c, keyPrefix, true)
```

Comment (ahrtr) [reply tests/robustness/traffic/kubernetes.go:106]: nit.

File: tests/robustness/traffic/kubernetes.go (hunk 2 above)
Comment (ahrtr) [reply tests/robustness/traffic/kubernetes.go:117]: minor comment: feel free to keep it as it's

Comment (serathius) [reply tests/robustness/traffic/kubernetes.go:117]: Prefer the old version, reason is that it better shows intention and it's less prone to code changes.

**Pull request merged**